pkg/controller/storagecluster/external_resources.go

```diff
@@ -123,25 +147,38 @@ func (r *ReconcileStorageCluster) createExternalStorageClusterResources(ownerRef
 		case "StorageClass":
 			var sc *storagev1.StorageClass
 			if d.Name == cephFsStorageClassName {
-				// Setting the fsname for cephfs StorageClass
+				// Setting the 'sc' point to CephFS StorageClass
 				sc = scs[0]
+				enableRookCSICephFS = true
 			} else if d.Name == cephRbdStorageClassName {
-				// Setting the PoolName for RBD StorageClass
+				// Setting the 'sc' point to RBD StorageClass
 				sc = scs[1]
 			} else if d.Name == cephRgwStorageClassName {
 				// Set the external rgw endpoint variable for later use on the Noobaa CR (as a label)
 				// Replace the colon with an underscore, otherwise the label will be invalid
 				externalRgwEndpointReplaceColon := strings.Replace(d.Data[externalCephRgwEndpointKey], ":", "_", -1)
 				externalRgwEndpoint = externalRgwEndpointReplaceColon
 
-				// Setting the Endpoint for OBC StorageClass
+				// Setting the 'sc' point to OBC StorageClass
 				sc = scs[2]
```

Comment:
I just realized something... `newStorageClasses()` doesn't always return three StorageClasses. The OBC StorageClass may not always be present. Should we have a safeguard in case there is a misconfiguration over the RGW StorageClass?

Reply:
@jarrpa , we don't require the check, why because the function `ensureExternalStorageClusterResources()` will only be called if we are in external mode (i.e; `Spec.ExternalStorage.Enable` is true) and `newStorageClasses()` function will allways return the third OBC storage class (`scs[2]`) without fail in external mode.

Reply:
Hm.... okay, I'll let it slide for now.